front/hooks/useFileUploaderService.ts

```diff
@@ -141,13 +145,65 @@ export function useFileUploaderService() {
       const results = await Promise.all(previewPromises);
 
       setFileBlobs([...fileBlobs, ...results]);
+
+      for (const fileBlob of results) {
```

Comment:
Should be a Promise.all here or at least have some form of parallelization as we def don't want to wait for a sequential upload?